components/auth/AuthProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { createContext, useState } from 'react';
+import { useContext } from 'react';
+
+type AuthContextData = {
+    authData: AuthData;
+    login(email: string, password: string): Promise<void>;
+    logout(): void;
+}
+
+type AuthData = {
+    username: string;
+    email: string;
+    password: string;
+}
+
+// create a context provider to hold auth data
+const AuthContext = createContext<AuthContextData>({} as AuthContextData);
+
+export const getAuthContext = () => {
+    const context = useContext(AuthContext);
+    return context;
+};
+
+export const AuthProvider: React.FC = ({ children }) => {
+
+    const [authData, setAuthData] = useState<any>();
```

Comment:
Yes, that's what I was trying to do. But it gives me an error in [line 43](https://github.com/che57/goodside-react-native-interview/blob/83dd56d7f7deb58dfb12fb736fc33f807186c588/components/auth/AuthProvider.tsx#L43), saying cannot assign AuthData | undefined to AuthData. It can be fixed simply by changing the type [here](https://github.com/che57/goodside-react-native-interview/blob/83dd56d7f7deb58dfb12fb736fc33f807186c588/components/auth/AuthProvider.tsx#L5) to AuthData | undefined. 

It didn't come to my mind when I implemented this part for some reasons LOL. So, I chose a drity way, just to let the app build and run.